### ISSUE TITLE
[DOC] inconsistent default value for `val_check_interval`

### ISSUE THREAD

**Issue by turbotimon** [Lightning-AI/pytorch-lightning#20678]
### 📚 Documentation

Inconsistent default value for `val_check_interval`, where the [Core API Documentation](https://lightning.ai/docs/pytorch/stable/common/trainer.html#val-check-interval) claims to be **`1.0`**:

> val_check_interval ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[int](https://docs.python.org/3/library/functions.html#int), [float](https://docs.python.org/3/library/functions.html#float), [None](https://docs.python.org/3/library/constants.html#None)]) – How often to check the validation set. Pass a float in the range [0.0, 1.0] to check after a fraction of the training epoch. Pass an int to check after a fixed number of training batches. An int value can only be higher than the number of training batches when check_val_every_n_epoch=None, which validates after every N training batches across epochs or during iteration-based training. Default: 1.0.

But the[ Trainer signature](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.trainer.trainer.Trainer.html#lightning.pytorch.trainer.trainer.Trainer) says **`None`**:

> classlightning.pytorch.trainer.trainer.Trainer(*, accelerator='auto', strategy='auto', devices='auto', num_nodes=1, precision=None, logger=None, callbacks=None, fast_dev_run=False, max_epochs=None, min_epochs=None, max_steps=-1, min_steps=None, max_time=None, limit_train_batches=None, limit_val_batches=None, limit_test_batches=None, limit_predict_batches=None, overfit_batches=0.0, val_check_interval=None, check_val_every_n_epoch=1, num_sanity_val_steps=None, log_every_n_steps=None, enable_checkpointing=None, enable_progress_bar=None, enable_model_summary=None, accumulate_grad_batches=1, gradient_clip_val=None, gradient_clip_algorithm=None, deterministic=None, benchmark=None, inference_mode=True, use_distributed_sampler=True, profiler=None, detect_anomaly=False, barebones=False, plugins=None, sync_batchnorm=False, reload_dataloaders_every_n_epochs=0, default_root_dir=None, model_registry=None)

Even tough, further down the same page, it says again *` Default: 1.0`*. This seems to be very confusing in situation where I e.g. only want to check every 2 batches by settings `check_val_every_n_epoch=2` because the `val_check_interval` would still be at its default value of `1.0` leading to confusion how often validation happens.

If I understand it correctly, when I explicitly set `check_val_every_n_epoch=2` it would take precedence over the `val_check_interval=1.0`? Therefore, "`Default: None`" makes more sense (also, it's the Class signature).

Having the default `1.0` in case I set explicit `check_val_every_n_epoch=None` also makes no sense, since the docs say `check_val_every_n_epoch: ... If None, validation will be done solely based on the number of training batches, requiring val_check_interval to be an integer value.`

**Comment by SkafteNicki** [comment]
@turbotimon thanks for raising this issue. Let me try to clarify: 
The reason that the documentation says that the default value is `1` but the trainer signature says `None` is that we internally converts `None->1` which is the reason we say that the value is `1`. This is the same for nearly all trainer flags, that the actual trainer signature is `None` but internally it gets converted to some other value.

To just clarify the interaction between `check_val_every_n_epoch` and `val_check_interval`:

1. **Primary control**: `check_val_every_n_epoch` determines if validation happens at epoch boundaries
2. **Secondary control**: `val_check_interval` determines intra-epoch validation frequency
3. **Precedence**: When `check_val_every_n_epoch` is set to an integer > 1, it gates the validation to only occur every N epochs, regardless of `val_check_interval`

The special case then happens when `check_val_every_n_epoch=None` 

- Validation is controlled entirely by `val_check_interval`
- `val_check_interval` must be an integer (not float) in this case
- Validation happens every N training batches across all epochs

I understand that this may be confusing, but the reason for having two parameters and that they interact with each others in this way is to make sure that we support all the different ways that users are running their training and we within these usecases support different ways to trigger validation.

**Comment by SkafteNicki** [comment]
Closing issue for now, please ping and reopen if necessary but for the time being the documentation does indicate the correct behavior that in practice the default value of `val_check_interval` is `1.0` even though the trainer signature says something else.

**Comment by turbotimon** [comment]
Hi @SkafteNicki sorry for not replying earlier. I think Lightning could do so much better by either setting it to literally `1.0` instead of `None` if this is really the default (there may be internal reasons to not do so, but it is hard to believe that this is not solvable) or mentioning it in the parameter description with `Default: None which translates to 1.0` (still not good, but better).

I have a lot of examples of the Lightning API, which are so confusing as you always have to look in three different places for the full answer:

- The function signature
- The function parameter description
- The tutorial or other places

**Comment by turbotimon** [comment]
An thank you for your explanation! I was looking for exact that in the documentation but couldn't find it that clearly explained. I needed to test it myself in cumpersome experiments to find that out...

> Primary control: check_val_every_n_epoch determines if validation happens at epoch boundaries
Secondary control: val_check_interval determines intra-epoch validation frequency
Precedence: When check_val_every_n_epoch is set to an integer > 1, it gates the validation to only occur every N epochs, regardless of val_check_interval
